src/packages/engine-core/src/NAPIEngine.ts

```diff
@@ -388,22 +391,28 @@ You may have to run ${chalk.greenBright(
 
   async stop(): Promise<void> {
     await this.connectPromise
-    debug('stop')
+    await this.currentQuery
```

Comment:
So now we wait for current query to finish?

Reply:
yep

Reply:
Are we currently doing that with NodeEngine? (I don't think so?)

Reply:
https://github.com/prisma/prisma/blob/client/telemetry/src/packages/engine-core/src/NodeEngine.ts#L840

Reply:
Nice thanks!